spec/jobs/reports/authentication_report_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'rails_helper'
+
+RSpec.describe Reports::AuthenticationReport do
+  let(:issuer) { 'issuer1' }
+  let(:issuers) { [issuer] }
+  let(:report_date) { Date.new(2023, 12, 25) }
+  let(:email)  { 'partner.name@example.com' }
+  let(:name) { 'Partner Name' }
+
+  let(:report_configs) do
+    [
+      {
+        'name' => name,
+        'issuers' => issuers,
+        'emails' => [email],
+      },
+    ]
+  end
+
+  before do
+    allow(IdentityConfig.store).to receive(:s3_reports_enabled).and_return(true)
+    allow(IdentityConfig.store).to receive(:weekly_auth_funnel_report_config) { report_configs }
+  end
+
+  describe '#perform' do
+    let(:tables) do
+      [
+        [
+          ['Some', 'String'],
+          ['a', 'b'],
+          ['c', 'd'],
+        ],
+        [
+          { float_as_percent: true, title: 'Custom Table 2' },
+          ['Float', 'Int', 'Float'],
+          ['Row 1', 1, 0.5],
+          ['Row 2', 1, 1.5],
+        ],
+        [
+          { float_as_percent: false, title: 'Custom Table 3' },
+          ['Float As Percent', 'Gigantic Int', 'Float'],
+          ['Row 1', 100_000_000, 1.0],
+          ['Row 2', 123_456_789, 1.5],
+        ],
+      ]
+    end
```

Comment:
I would stub with a table that looks a little more like the value returned by `AuthenticationReport`? as-is, that class won't have the first metadata row, and is only a single table

Reply:
updated in https://github.com/18F/identity-idp/pull/9295/commits/963af3d77d24b7d746c46dcfc229203de89c890f